src/aspire/utils/blk_diag_func.py

```diff
@@ -140,50 +239,123 @@ def mat_to_blk_diag(mat, blk_partition):
 
 
 def blk_diag_transpose(blk_diag):
+    """
+    Get the transpose matrix of a block diagonal matrix
+
+    :param blk_diag: The block diagonal matrix
+    :return: The corresponding transpose form of a block diagonal matrix
+    """
+
     blk_diag_t = []
     for i in range(0, len(blk_diag)):
         blk_diag_t.append(blk_diag[i].T)
     return blk_diag_t
 
 
 def blk_diag_compatible(blk_diag_a, blk_diag_b):
+    """
+    Check whether two block diagonal matrices are compatible or not
+
+    :param blk_diag_a: The block diagonal A matrix
+    :param blk_diag_b: The block diagonal B matrix
+    :return: `True` value if all of shapes in two list representations of
+        block diagonal matrices are the same, otherwise return `False`.
+    """
     if len(blk_diag_a) != len(blk_diag_b):
         return False
-    elif np.allclose([np.size(mat_a, 1) for mat_a in blk_diag_a], [np.size(mat_b, 0) for mat_b in blk_diag_b]):
+    elif np.allclose([np.size(mat_a, 1) for mat_a in blk_diag_a],
+                     [np.size(mat_b, 0) for mat_b in blk_diag_b]):
         return True
     else:
         return False
 
 
 def blk_diag_isnumeric(x):
+    """
+    Check a block diag matrix is numeric or not
+    """
     try:
         return 0 == x*0
     except:
         return False
 
 
+def nonradial_filter2fb_mat(h_fun, fbasis):
+    """
+    Convert a nonradial function in k space into a basis representation
+
+    :param h_fun: The function form in k space
+    :param fbasis: The basis object for expanding
+    :return: a matrix representation using the `fbasis` expansion
+    """
+    if not isinstance(fbasis, FFBBasis2D):
+            raise NotImplementedError('Currently only fast FB method is supported')
+    # Set same dimensions as basis object
+    n_k = int(np.ceil(4 * fbasis.rcut * fbasis.kcut))
+    n_theta = np.ceil(16 * fbasis.kcut * fbasis.rcut)
+    n_theta = int((n_theta + np.mod(n_theta, 2)) / 2)
+    # get 2D grid in polar coordinate
+    k_vals, wts = lgwt(n_k, 0, 0.5)
+    k, theta = np.meshgrid(k_vals, np.arange(n_theta) * 2 * np.pi / (2 * n_theta), indexing='ij')
+    # Get function values in polar 2D grid and average out angle contribution
+    omegax = k*np.cos(theta)
+    omegay = k*np.sin(theta)
+    omega = 2 * np.pi * np.vstack((omegax.flatten('C'), omegay.flatten('C')))
+    h_vals2d = h_fun(omega).reshape(n_k, n_theta)
+    h_vals = np.sum(h_vals2d, axis=1)/n_theta
+
+    # Represent 1D function values in fbasis
+    return fun2fb_mat(k_vals, wts, h_vals, fbasis)
+
+
 def radial_filter2fb_mat(h_fun, fbasis):
```

Comment:
Why a separate function? Shouldn't the `nonradial_filter2fb_mat` give the same result?

Reply:
Combined into one function as in new commit 1ed1cca.